### PR TITLE
No longer require the user to supply a pre-existing parametric image

### PR DESCRIPTION
**Pull request by rijobro**
the changes concern `apply_patlak_to_images`

- If the output file already exists (first argument), use it as a template (same behaviour as before, doesn't break backwards compatibility)
- If the output does not exist, create the template based on the dynamic image

### CODE REVIEW

File: src/modelling_buildblock/ParametricDiscretisedDensity.cxx [hunk 1]
```diff
@@ -53,6 +54,25 @@ get_num_params()
   return dummy.size();
 }
 
+TEMPLATE
+ParamDiscDensity::
+ParametricDiscretisedDensity(const DynamicDiscretisedDensity& dyn_im)
+    : base_type(dyn_im[1].get_index_range(),
```
[the remark this answers is not in the document]
Comment (rijobro) [reply src/modelling_buildblock/ParametricDiscretisedDensity.cxx]: ok done (with changes for 342 as part of this PR, too)

File: src/include/stir/DynamicDiscretisedDensity.h [hunk 2]
```diff
@@ -162,6 +162,14 @@ class DynamicDiscretisedDensity: public ExamData
 
   const float get_calibration_factor() const;
 
+  //! at method
+  const singleDiscDensT & at(const unsigned int frame_num) const
+  { return this->get_density(frame_num); }
+
+  //! at method
+  const singleDiscDensT & at(const unsigned int frame_num)
```
Comment (KrisThielemans) [inline src/include/stir/DynamicDiscretisedDensity.h]: euh. Either we don't have this method, or we return a non-`const` ref. I guess the latter as that's what we have for `get_density`

Comment (rijobro) [reply src/include/stir/DynamicDiscretisedDensity.h]: apologies - typo. now fixed

**Comment by KrisThielemans** [comment]
Travis fails with something like
```
/home/travis/build/UCL/STIR/src/modelling_utilities/apply_patlak_to_images.cxx: In function ‘int main(int, char**)’:
/home/travis/build/UCL/STIR/src/modelling_utilities/apply_patlak_to_images.cxx:103:28: error: ‘make_shared’ is not a member of ‘std’
           par_image_sptr = std::make_shared<ParametricVoxelsOnCartesianGrid>(dyn_image);
                            ^
/home/travis/build/UCL/STIR/src/modelling_utilities/apply_patlak_to_images.cxx:103:28: note: suggested alternative:
In file included from /usr/include/boost/smart_ptr/make_shared.hpp:18:0,
                 from /usr/include/boost/make_shared.hpp:15,
                 from /home/travis/build/UCL/STIR/src/include/stir/shared_ptr.h:35,
                 from /home/travis/build/UCL/STIR/src/include/stir/IO/ExamData.h:27,
                 from /home/travis/build/UCL/STIR/src/include/stir/DiscretisedDensity.h:44,
                 from /home/travis/build/UCL/STIR/src/include/stir/modelling/ParametricDiscretisedDensity.h:31,
                 from /home/travis/build/UCL/STIR/src/modelling_utilities/apply_patlak_to_images.cxx:49:
/usr/include/boost/smart_ptr/make_shared_array.hpp:24:5: note:   ‘boost::make_shared’
     make_shared(std::size_t size) {
```
I guess we need to use the `MAKE_SHARED` work-around or so for pre-C++11. sorry

**Comment by rijobro** [comment]
Done, just waiting on travis.

**Pull request merged**